### PR TITLE
KAFKA-3421: Update docs with new connector features

### PR DESCRIPTION
**Pull request by Ishiihara**
@ewencp @gwenshap Docs. I also tried to clean up some typos. However, it seems that the we don't have two words without space in between in the source yet they showed up as no space in between in the generated doc.

### CODE REVIEW

File: docs/connect.html [hunk 1]
```diff
@@ -25,7 +25,7 @@
     <li><b>Distributed and standalone modes</b> - scale up to a large, centrally managed service supporting an entire organization or scale down to development, testing, and small production deployments</li>
     <li><b>REST interface</b> - submit and manage connectors to your Kafka Connect cluster via an easy to use REST API</li>
     <li><b>Automatic offset management</b> - with just a little information from connectors, Kafka Connect can manage the offset commit process automatically so connector developers do not need to worry about this error prone part of connector development</li>
-    <li><b>Distributed and scalable by default</b> - Kafka Connect builds on the existing </li>
+    <li><b>Distributed and scalable by default</b> - Kafka Connect builds on the existing group management protocol. More workers can be added to scale up Kafka Connect cluster.</li>
```
Comment (ewencp) [inline docs/connect.html]: to scale up Kafka Connect cluster -> to scale up a Kafka Connect cluster

File: docs/connect.html [hunk 2]
```diff
@@ -47,7 +47,7 @@
 
 The remaining parameters are connector configuration files. You may include as many as you want, but all will execute within the same process (on different threads).
 
-Distributed mode handles automatic balancing of work, allows you to scale up (or down) dynamically, and offers fault tolerance both in the active tasks and for configuration and offset commit data. Execution is very similar to standalone mode:
+Distributed mode handles automatic work balancing work, allows you to scale up (or down) dynamically, and offers fault tolerance both in the active tasks and for configuration and offset commit data. Execution is very similar to standalone mode:
```
Comment (ewencp) [inline docs/connect.html]: Hmm, seems we've introduced a typo -- "work balancing work". What was wrong with the original version of this sentence?

**Comment by ewencp** [comment]
@Ishiihara It looks like KAFKA-3461 might have already caught the typos I had mentioned to you, I just missed that commit when it happened.

File: docs/connect.html [hunk 3]
```diff
@@ -98,6 +99,13 @@
     <li><code>DELETE /connectors/{name}</code> - delete a connector, halting all tasks and deleting its configuration</li>
 </ul>
 
+We also provides a REST API for getting information of connector plugins:
```
Comment (ewencp) [inline docs/connect.html]: We -> Connect

Comment (ewencp) [inline docs/connect.html]: information of -> information about

File: docs/connect.html [hunk 4]
```diff
@@ -98,6 +99,13 @@
...
+
+<ul>
+    <li><code>GET /connector-plugins</code><- return a list of connector plugins installed in the Kafka Connect cluster</li>
```
Comment (ewencp) [inline docs/connect.html]: extra `<` character after closing the `</code>` tag

Comment (ewencp) [inline docs/connect.html]: Not sure if this needs to be documented or not, but this only checks for connectors on the worker that handles the request. This means you can actually see inconsistent results (and should during a rolling upgrade if you add new connector jars)

File: docs/connect.html [hunk 5]
```diff
@@ -98,6 +99,13 @@
...
+    <li><code>PUT /connector-plugins/{connector-type}/config/validate</code> - validate the provided configuration values against the configuration definition</li>
```
Comment (ewencp) [inline docs/connect.html]: Again, we give very limited docs here, but might want to clarify how this works a bit. You can run validation, get back per-field error info, suggested values, etc.

File: docs/connect.html [hunk 6]
```diff
@@ -183,6 +191,9 @@
 }
 </pre>
 
+Although not used in the example, <code>SourceTask</code> also provides two APIs to commit offsets in the source system: <code>commit</code> and <code>commitSourceRecord</code>. The APIs are provided for source systems that also need to store offsets internally which serve as an acknowledge mechanism to the source system that the message are delivered to Kafka.
```
Comment (ewencp) [inline docs/connect.html]: Second sentence is at least missing a comma (between "internally" and "which").

But I might rephrase this to make the use case a bit clearer. e.g. something like, "The APIs are provided for source systems which have an acknowledgement mechanism for messages. Overriding these methods allows the source connector to acknowledge messages in the source system, either in bulk or individually, once they have been successfully written to Kafka."

File: docs/connect.html [hunk 7]
```diff
@@ -183,6 +191,9 @@
...
+The <code>commit<code> API stores the offsets in the source system, up to the offsets that have been returned by <code>poll</code>. The implementation of this API should block until the commit is complete. The <code>commitSourceRecord</code> API saves the offset in the source system for each <code>SourceRecord</code> after it is delivered to Kafka.As Kafka Connect will record offsets automatically, <code>SourceTask<code>s are not required to implement them. Typically you only need to implement one of the two APIs.
```
Comment (ewencp) [inline docs/connect.html]: Missing space between sentences: Kafka.As -> Kafka. As

For the final sentence, since you just said most won't need to implement them, I'd start out with something like "In cases where a connector does need to acknowledge messages in the source system, only one of the APIs is typically required."

**Comment by ewencp** [comment]
@Ishiihara I had a few suggestions on tweaking wording, but mostly looks good.

Since we don't have pause/resume/restart yet, we may need a follow up JIRA if we don't get those committed and included in this patch. Note that the restart patch is landing right now (but w/o docs), and pause/resume should follow soon.

**Comment by Ishiihara** [comment]
@ewencp Addressed review comments. Will file another JIRA with the pause/resume/restart APIs. https://issues.apache.org/jira/browse/KAFKA-3583

**Comment by ewencp** [comment]
LGTM, thanks!
